plugins/modules/cloud/opennebula/one_template.py

```diff
@@ -0,0 +1,263 @@
+      - managed by this module.
+    type: str
+  template:
+    description:
+      - A string containing the template contents.
+    type: str
+  state:
+    description:
+      - C(present) - state that is used to manage the template
+      - C(absent) - delete the template
+    choices: ["present", "absent"]
+    default: present
+    type: str
+
+extends_documentation_fragment:
+- community.general.opennebula
+
+author:
+    - "Georg Gadinger (@nilsding)"
+'''
+
+EXAMPLES = '''
+- name: Fetch the TEMPLATE by id
+  community.general.one_template:
+    id: 6459
+  register: result
+
+- name: Print the TEMPLATE properties
+  ansible.builtin.debug:
+    var: result
+
+- name: Fetch the TEMPLATE by name
+  community.general.one_template:
+    name: tf-prd-users-workerredis-p6379a
+  register: result
+
+- name: Create a new or update an existing TEMPLATE
+  community.general.one_template:
+    name: generic-opensuse
+    template: |
+      CONTEXT = [
+        HOSTNAME = "generic-opensuse"
+      ]
+      CPU = "1"
+      CUSTOM_ATTRIBUTE = ""
+      DISK = [
+        CACHE = "writeback",
+        DEV_PREFIX = "sd",
+        DISCARD = "unmap",
+        IMAGE = "opensuse-leap-15.2",
+        IMAGE_UNAME = "oneadmin",
+        IO = "threads",
+        SIZE = "" ]
+      MEMORY = "2048"
+      NIC = [
+        MODEL = "virtio",
+        NETWORK = "testnet",
+        NETWORK_UNAME = "oneadmin" ]
+      OS = [
+        ARCH = "x86_64",
+        BOOT = "disk0" ]
+      SCHED_REQUIREMENTS = "CLUSTER_ID=\\"100\\""
+      VCPU = "2"
+
+- name: Delete the TEMPLATE by id
+  community.general.one_template:
+    id: 6459
+    state: absent
+'''
+
+RETURN = '''
+id:
+    description: template id
+    type: int
+    returned: success
+    sample: 153
+name:
+    description: template name
+    type: str
+    returned: success
+    sample: app1
+template:
+    description: the parsed template
+    type: dict
+    returned: success
+group_id:
+    description: template's group id
+    type: int
+    returned: success
+    sample: 1
+group_name:
+    description: template's group name
+    type: str
+    returned: success
+    sample: one-users
+owner_id:
+    description: template's owner id
+    type: int
+    returned: success
+    sample: 143
+owner_name:
+    description: template's owner name
+    type: str
+    returned: success
+    sample: ansible-test
+'''
+
+
+from ansible_collections.community.general.plugins.module_utils.opennebula import OpenNebulaModule
+
+
+class TemplateModule(OpenNebulaModule):
+    def __init__(self):
+        argument_spec = dict(
+            id=dict(type='int', required=False),
+            name=dict(type='str', required=False),
+            state=dict(type='str', choices=['present', 'absent'], default='present'),
+            template=dict(type='str', required=False),
+        )
+
+        mutually_exclusive = [
+            ['id', 'name']
+        ]
+
```

Comment:
I haven checked how to handle that in `OpenNebulaModule`, but I think it would make sense to use:
```yaml
required_one_of = [('id', 'name']
```
So that the module will complain if none of those is passed.

Reply:
I'll give that a try, thanks